init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java

```diff
@@ -18,30 +20,55 @@
     private final RoleServiceInvoker roleServiceInvoker;
     private final SiteServiceInvoker siteServiceInvoker;
     private final Trial trialConfig;
+    private final InitProgressService initProgressService;
 
     @Value("${delay-start:1}")
     private long delayStartInSeconds;
 
     @Autowired
     public Loader(Trial trialConfig, PractitionerServiceInvoker practitionerServiceInvoker,
-                  RoleServiceInvoker roleServiceInvoker, SiteServiceInvoker siteServiceInvoker) {
+                  RoleServiceInvoker roleServiceInvoker, SiteServiceInvoker siteServiceInvoker,
+                  InitProgressService initProgressService) {
```

Comment:
initProgress 'service' sounds like any of the other services:
sites service, role service , practitioner service.

maybe initProgressReporter? initProgressStatus? or something like this

Reply:
Totally agree with you, it can mislead, I gave this name because it's a service under "service" folder,
initProgressReporter sounds much better, I'll rename it.